dependencies/dependencies.go

```diff
@@ -27,23 +29,31 @@ func (d DefaultDependencies) HTTPClient() (*http.Client, error) {
 	return nil, errors.New(codes.Invalid, "http client uninitialized in dependencies")
 }
 
+func (d DefaultDependencies) SecretService() (SecretService, error) {
+	if d.secretservice != nil {
+		return d.secretservice, nil
+	}
+	return nil, errors.New(codes.Invalid, "secret service uninitialized in dependencies")
```

Comment:
Small thing: `codes.Invalid` doesn't seem appropriate. I think `codes.Unimplemented` is correct.